include/zephyr/sys/util.h

```diff
@@ -783,6 +784,46 @@ static inline void mem_xor_128(uint8_t dst[16], const uint8_t src1[16], const ui
 	mem_xor_n(dst, src1, src2, 16);
 }
 
+/**
+ * @brief Compare memory areas and their length
+ *
+ * If the length are 0, return true.
+ *
+ * @param m1 First memory area to compare, cannot be NULL even if length is 0
+ * @param len1 Length of the first memory area to compare
+ * @param m2 Second memory area to compare, cannot be NULL even if length is 0
+ * @param len2 Length of the second memory area to compare
+ *
+ * @returns true if both the length of the memory areas and their content are
+ * equal else false
+ */
+static inline bool memeq(const void *m1, size_t len1, const void *m2, size_t len2)
+{
+	int ret;
+
+	if (len1 != len2) {
+		return false;
+	}
+
+	ret = memcmp(m1, m2, len1);
+
+	return ret == 0;
+}
```

Comment:
```suggestion
static inline bool memeq(const void *m1, size_t len1, const void *m2, size_t len2)
{
	return len1 == len2 && bool_memcmp(m1, m2, len1);
}
```

Suggest a less verbose version of this function